crates/ibc/src/context/nft_transfer.rs

```diff
@@ -0,0 +1,372 @@
+//! IBC Non-Fungible token transfer context
+
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use namada_core::ibc::apps::nft_transfer::context::{
+    NftTransferExecutionContext, NftTransferValidationContext,
+};
+use namada_core::ibc::apps::nft_transfer::types::error::NftTransferError;
+use namada_core::ibc::apps::nft_transfer::types::{
+    ClassData, ClassUri, Memo, PrefixedClassId, TokenData, TokenId, TokenUri,
+    PORT_ID_STR,
+};
+use namada_core::ibc::core::handler::types::error::ContextError;
+use namada_core::ibc::core::host::types::identifiers::{ChannelId, PortId};
+use namada_core::types::address::Address;
+use namada_core::types::ibc::{NftClass, NftMetadata, IBC_ESCROW_ADDRESS};
+use namada_core::types::token::Amount;
+
+use super::common::IbcCommonContext;
+use crate::storage;
+
+/// NFT transfer context to handle tokens
+#[derive(Debug)]
+pub struct NftTransferContext<C>
+where
+    C: IbcCommonContext,
+{
+    inner: Rc<RefCell<C>>,
+}
+
+impl<C> NftTransferContext<C>
+where
+    C: IbcCommonContext,
+{
+    /// Make new NFT transfer context
+    pub fn new(inner: Rc<RefCell<C>>) -> Self {
+        Self { inner }
+    }
+
+    /// Update the mint amount of the token
+    fn update_mint_amount(
+        &self,
+        token: &Address,
+        is_minted: bool,
+    ) -> Result<(), NftTransferError> {
+        let mint = self.inner.borrow().mint_amount(token)?;
+        let updated_mint = if is_minted {
+            mint.checked_add(Amount::from_u64(1)).ok_or_else(|| {
+                NftTransferError::Other(
+                    "The mint amount overflowed".to_string(),
+                )
+            })?
```

Comment:
realistically this error should be impossible, no? we can only hold 1 unit at a time. we should be checking that `mint` is equal to 0, then return `Amount::from_u64(1)` or smth